File: exercism/csharp/bob/Bob.cs
using System.Text.RegularExpressions;


namespace Exercism
{
    public static class Bob
    {
        public static string Hey(string prompt)
        {
            if (Yelling(prompt)) return "Whoa, chill out!";
            if (Question(prompt)) return "Sure.";
            if (Silence(prompt)) return "Fine. Be that way!";
            return "Whatever.";
        }

        private static bool Yelling(string prompt)
        {
            var alpha = new Regex("[a-zA-Z]");
            return alpha.IsMatch(prompt) && prompt == prompt.ToUpper();
        }

        private static bool Question(string prompt)
        {
            var s = prompt.Trim();
            return s.Length > 0 && s[s.Length - 1] == '?';
        }

        private static bool Silence(string prompt)
        {
            return prompt.Trim() == "";
        }
    }
}

File: exercism/cpp/bob/bob.cpp
#include <regex>
#include "bob.h"

// At least one upper case letter, no lower case letters
std::regex yelling("^[^a-z]+[A-Z][^a-z]+$");

// Question mark at the end
std::regex question("\\?\\W*$");

// Nothing but whitespace
std::regex silence("^\\W*$");

std::string bob::hey(const std::string &prompt) {
  if (regex_search(prompt, yelling))
    return "Whoa, chill out!";
  if (regex_search(prompt, question))
    return "Sure.";
  if (regex_search(prompt, silence))
    return "Fine. Be that way!";
  return "Whatever.";
}
